File: rb/features.rb
# Tempmail SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module TempmailFeatures
  def self.make_feature(name)
    case name
    when "base"
      TempmailBaseFeature.new
    when "test"
      TempmailTestFeature.new
    else
      TempmailBaseFeature.new
    end
  end
end

File: py/features.py
# Tempmail SDK feature factory

from feature.base_feature import TempmailBaseFeature
from feature.test_feature import TempmailTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TempmailBaseFeature(),
        "test": lambda: TempmailTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
